workflows/rnaseq-ref-index/make_pre_mrna_fasta.R

```diff
@@ -123,14 +140,46 @@ eisaR::exportToGtf(
   filepath = spliced_intron_gtf
 )
 
-# create text file mapping transcript and intron identifiers to 
-# corresponding gene identifiers
-# first make Tx2Gene for all spliced and intron sequences
+# create text file mapping transcript and intron identifiers to corresponding gene identifiers
+# make 2 column Tx2Gene for all spliced and intron sequences
 full_tx2gene <- eisaR::getTx2Gene(
   grl, filepath = spliced_intron_tx2gene
 )
 
-## need to write out to metadata 
+# trim grl used for splici
+splici_grl <- trim(grl)
```

Comment:
What does trim do here? Is this the main difference remaining for splici and the previous grl (besides the later unique)?

The question here is if this is something we should just use for both...

Reply:
Hi @allyhawkins and @jashapiro --- sorry for not checking in on this the past couple of days.  The point if `trim` is that it prevents an out of bounds error if the intron plus the chosen flank extends past the end of the contig.  We found that sometimes there are a handful (usually just one or two) genes that are close to the end of a chromosome, and if you choose too long a flank length, the GenomicRange returned will overhang the chromosome.  This in turn causes an indexing error and the whole extraction procedure to fail.  The call to `trim()` prevents this, clamping intervals to start no earlier than the chromosome start and end no later than its length.